tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -19,46 +19,95 @@ def decode_name_with_port(input_model: InputModel, node_name: str, framework="")
-        found_nodes.append(node)
-
-
-    def try_get_node(model, name, framework):
-        node = model.get_place_by_operation_name(name)
-        if node:
-            return node
-        if framework == "onnx":
-            tensor = model.get_place_by_tensor_name(name)
-            if tensor:
-                return tensor.get_producing_operation()
-        return None
-
-
-    regexp_post = r'(.+):(\d+)'
-    match_post = re.search(regexp_post, node_name)
-    if match_post:
-        node_post = try_get_node(input_model, match_post.group(1), framework)
-        if node_post:
-            node_post = node_post.get_output_port(output_port_index=int(match_post.group(2)))
-            if node_post:
-                found_node_names.append(match_post.group(1))
-                found_nodes.append(node_post)
-
-    regexp_pre = r'(\d+):(.+)'
-    match_pre = re.search(regexp_pre, node_name)
-    if match_pre:
-        node_pre = try_get_node(input_model, match_pre.group(2), framework)
-        if node_pre:
-            node_pre = node_pre.get_input_port(input_port_index=int(match_pre.group(1)))
-            if node_pre:
-                found_node_names.append(match_pre.group(2))
-                found_nodes.append(node_pre)
 
+    def try_get_nodes(input_model, node_name):
+        # Passed node_name can be in several forms:
+        # 1) name
+        # 2) port:name
+        # 3) name:port
+        found_nodes = []
+        found_node_names = []
+        # if we find tensor, there is no need
+        # to continue searching
+        search_tensor = True
+        # check if there is a tensor with
+        # given node_name
+        tensor = input_model.get_place_by_tensor_name(node_name)
+        if tensor:
+            found_node_names.append('Tensor:' + tensor.get_names()[0])
+            found_nodes.append(tensor)
+            search_tensor = False
+
+        # first we check for port:name combination
+        regexp_pre = r'(\d+):(.+)'
+        match_pre = re.search(regexp_pre, node_name)
+        # check if there is a match
+        if match_pre:
+            # split node_name into port and name
+            port, name = match_pre.group(1), match_pre.group(2)
+            # check if there is an operation with name
+            node = input_model.get_place_by_operation_name(name)
+            # if there is no operation with provided name and we
+            # did not find tensor before we check if there is a
+            # tensor with name
+            if not node and search_tensor:
+                tensor = input_model.get_place_by_tensor_name(name)
+                if tensor:
+                    found_node_names.append('Tensor:' + tensor.get_names()[0])
+                    found_nodes.append(tensor)
+                    search_tensor = False
+            if node:
+                # if there is an operation with given name, we
+                # add input port
+                node_pre = node.get_input_port(input_port_index=int(port))
```

Comment:
what if port is `None`? what input port is retrieved then?

Reply:
if line 62 is being called, then port cannot be None, since to get there there must be regex match with `node_name` which extracts port:name